pkg/cloud/services/ec2/subnets.go

```diff
@@ -146,6 +131,51 @@ LoopExisting:
 	return nil
 }
 
+func (s *Service) getDefaultSubnets() (infrav1.Subnets, error) {
+	zones, err := s.getAvailableZones()
+	if err != nil {
+		return nil, err
+	}
+
+	//TODO: should this be configurable or are we ok with 3 AZs max for the default?
+	if len(zones) > 3 {
+		s.scope.V(2).Info("region has more than 3 availability zones, picking 3 at random", "region", s.scope.Region())
+		rand.Shuffle(len(zones), func(i, j int) {
+			zones[i], zones[j] = zones[j], zones[i]
+		})
+		zones = append(zones, zones[0], zones[1], zones[2])
+	}
+
+	// 1 private subnet for each AZ plus 1 other subnet that will be further sub-divided for the public subnets
+	numSubnets := len(zones) + 1
+	subnetCIDRs, err := cidr.SplitIntoSubnetsIPv4(s.scope.VPC().CidrBlock, numSubnets)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed splitting VPC CIDR %s into subnets", s.scope.VPC().CidrBlock)
+	}
+
+	publicSubnetCIDRs, err := cidr.SplitIntoSubnetsIPv4(subnetCIDRs[0].String(), len(zones))
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed splitting CIDR %s into public subnets", subnetCIDRs[0].String())
+	}
+	privateSubnetCIDRs := append(subnetCIDRs[:0], subnetCIDRs[1:]...)
```

Comment:
Should we validate that the CidrBlock has at least a minimum number of IP addresses to avoid splitting it too much and have very few IPs per subnet.

What's the default CIDR here for the VPC? Can we also document it somewhere?

Reply:
The default CIDR block hasn't been changed and is `10.0.0.0/16`. I can add this to the documents.

Thats a very good point about not having enough IPs per subnet.  From the AWS docs CIDR blocks must be in the range: ` /16 netmask (65,536 IP addresses) and /28 netmask (16 IP addresses)` , so if a /28 is used and then we try and split into 3 subnets then each subnet is only going to have less than a handful of IPs which isn't enough. I can add a check for this and report an error

Reply:
Even 16 IPs is very low, @randomvariable what do you think should be the bare minimum here?

Reply:
/24 is still fine. Don't think you want to go too far below that number of IP addresses. In particular, that gives enough "breathing space" for ELBv1 to consume IP space (need ~30 free IP addresses minimum otherwise it starts to have issues).

Have previously used /17 for the VPC size as a compromise for users who may have on-prem networks they need to connect up.

Reply:
I will add a check for the VPC CIDR block. 

If its above /24 should i error or allow with a warning?

Reply:
@vincepri @randomvariable - should i add a check for the CIDR block as part of this change? Or shall we create another issue, which could be a 'good first issue'?

Reply:
Don't do any validation, was more speaking about defaults. I think people will know what they're doing if they go > /24 and will have a specific case for it.